enhancements/ingress/admin-configured-ca-bundle.md

```diff
@@ -0,0 +1,314 @@
+  certificate when a destination CA certificate is not configured for a route.
+- As an OpenShift cluster administrator, I want OpenShift router to update its configuration whenever the
+  OpenShift service CA certificate or the admin-configured CA certificate bundle is updated or deleted.
+- As an OpenShift cluster administrator, I want OpenShift router to verify services' certificates with just
+  the configured destination CA Certificate and not the default destination CA certificate.
+
+### Goals
+
+- An OpenShift administrator can configure a CA certificate bundle and expect OpenShift router to use it
+  for verifying a service's certificate when destination CA Certificate isn't specified in the route object
+  created for the service.
+- Openshift router should seamlessly reload the OpenShift service CA certificate or the
+  administrator-configured CA certificate bundle when either is modified.
+
+### Non-Goals
+
+- OpenShift administrator defined CA certificate bundle will be used only for the traffic through OpenShift
+  router and for the routes created with re-encrypt termination type, any other scenario is outside the
+  scope of this proposal.
+- The common CA bundle created by the operator is for all ingress controllers usage, and defining a CA
+  bundle specific for each IngressController usage is out of scope.
+
+## Proposal
+
+Change OpenShift Ingress Operator to:
+1. Create a CA trust bundle which will contain the OpenShift service CA certificate and the custom
+   user-supplied CA certificates.
+2. Make the new common certificate available to the OpenShift Router as the default destination CA bundle
+   file to be used for verifying the services' certificate.
+3. Updates the CA bundle file whenever either of the certificates change.
+
+Update router to:
+1. Watch the common CA bundle and update its configuration internally when needed.
+
+### Workflow Description
+
+- An OpenShift administrator configures the list of trusted CA certificates as a ConfigMap, named
+  specifically `admin-ca-bundle` in `openshift-config` namespace.
+- OpenShift Ingress Operator creates `ingress-ca-bundle` ConfigMap in `openshift-ingress` namespace by
+  concatenating the admin configured CA certificates and the OpenShift service CA certificate stored in
+  the respective ConfigMaps.
+- OpenShift Ingress Operator creates a watcher and reconciles whenever changes are observed in the
+  aforementioned ConfigMap objects.
+- The Ingress Operator configures OpenShift router with a ConfigMap volume mount to use the operator-created ConfigMap as the default destination CA certificate
+  to verify a service's certificate when a route with re-encrypt termination type is created with an empty
+  destination CA.
+- OpenShift router watches the volume mount and reloads HAProxy whenever the CA certificate bundle changes.
+
+![alt Workflow diagram](admin-configured-ca-bundle.png)
+
+#### Variation [optional]
```

Comment:
having reached this point.  if https://github.com/kubernetes/api/blob/39c80441f8144164dc6c2b815cbeb8564ed31e6d/certificates/v1alpha1/types.go#L28-L43 were enabled, would we be talking about having a route indicate, "use this clustertrustbundle.spec.signerName to verify me" on a field and the cluster-admin (or other) would managed the ClusterTrustBundle?

Reply:
Please correct me, if my understanding of the API is correct, this would require the user to make use of CertificateSigningRequests for obtaining the services' certificates, but for the routes usecase, the signer need not be made available in openshift and the service can have key pair assigned which were generated externally correct. 
If `ClusterTrustBundle` would be used, then it would be signer-unlinked types I think.